docs/example-findByText.md

```diff
@@ -0,0 +1,178 @@
+    <label for='username'>
+      User Name<span id='username_required_error' style='color: red; display: none;'> required</span>:
+    </label>
+
+    <input
+      type='text'
+      name='username'
+      id='username_input'
+      placeholder='Enter user name'
+    />
+
+    <span id='invalid_username_error' style='color: red; display: none;'>
+      Invalid User Name
+    </span>
+
+    <label for='password'>
+      Password<span id='password_required_error' style='color: red; display: none;'> required</span>:
+    </label>
+
+    <input
+      type='password'
+      name='password'
+      id='password_input'
+      placeholder='Enter password'
+    />
+
+    <span id='invalid_password_error' style='color: red; display: none;'>
+      Invalid Password
+    </span>
+
+    <button id='submit' type='submit'>
+      Login
+    </button>
+  </form>
+  `;
+
+  const submitButton = el.querySelector("#submit");
+  const formEl = el.querySelector("#login_form");
+
+  submitButton.addEventListener("click", () => {
+    const userInput = el.querySelector("#username_input");
+    const passwordInput = el.querySelector("#password_input");
+
+    var userName = userInput.value;
+    var password = passwordInput.value;
+    if (!userName) {
+      el.querySelector("#username_required_error").style.display = "inline";
+    }
+
+    if (!password) {
+      el.querySelector("#password_required_error").style.display = "inline";
+    }
+
+    if (userName && userName !== "Bob") {
+      el.querySelector("#invalid_username_error").style.display = "inline";
+    }
+
+    if (password && password !== "theBuilder") {
+      el.querySelector("#invalid_password_error").style.display = "inline";
+    }
+  });
+
+  formEl.addEventListener("submit", function (evt) {
+    evt.preventDefault();
+    window.history.back();
+  });
+
+  return el;
+};
+
+describe("findBy* Examples", () => {
+  let div;
+  let container;
+
+  beforeEach(() => {
+    div = document.createElement("div");
+    container = renderContent(div);
+  });
+
+  it("should show a required field warning for each empty input field", async () => {
+    expect(
+      await findByText(container, (content, element) => {
+        return (
+          element.tagName.toLowerCase() === "label" && content === "User Name:"
+        );
+      })
+    ).toBeVisible();
+    expect(
+      await findByText(container, (content, element) => {
+        return (
+          element.tagName.toLowerCase() === "label" && content === "Password:"
+        );
+      })
+    ).toBeVisible();
+
+    userEvent.click(
+      getByRole(container, "button", {
+        name: "Login"
+      })
+    );
+
+    expect(
+      // https://stackoverflow.com/questions/55509875/how-to-query-by-text-string-which-contains-html-tags-using-react-testing-library
+      await findByText(container, (_, element) => {
+        const hasText = (element) =>
+          element.textContent.replace(/(\r\n|\n|\r)/gm, "").trim() ===
```

Comment:
I formatted the `html` in `renderContent` and apparently `textContent` will preserve line breaks and new lines, I've used this regex to remove it. Let me know if there's a better way to handle this!

I also used the custom text matcher function from the above StackOverflow answer.

Reply:
I would try to keep the incidental complexity in the example to a minimum; focus on showing the key info about the findBy queries. Is there a way you can trim it down?

Reply:
Thanks, @alexkrolick, that's a good point - I've changed the example so the warning shows the same way that the "invalid user name/password" text does to avoid inline HTML elements to keep the complexity down! Let me know if there's anything else that needs fixing up.